clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -956,14 +955,71 @@ public String toRst() {
             b.append("  * Importance: ");
             b.append(def.importance.toString().toLowerCase(Locale.ROOT));
             b.append("\n\n");
+        }
+        return b.toString();
+    }
+
+    /**
+     * Get the configs with new metadata(group, orderInGroup, dependents) formatted with reStructuredText, suitable for embedding in Sphinx
+     * documentation.
+     */
+    public String toEnrichedRst() {
+        List<ConfigKey> configs = sortedConfigsInGroup();
+        StringBuilder b = new StringBuilder();
+        String lastKeyGroupName = "";
 
-            b.append("  * Dependents: ");
-            for (String dependent : def.dependents)
-            {
-                b.append(dependent);
-                b.append("\t");
+        for (ConfigKey def : configs) {
+
+            if (!def.group.equalsIgnoreCase(lastKeyGroupName)) {
+                b.append(def.group);
```

Comment:
need a newline after the group before the underline, and a double newline after the underline